compiler/crates/relay-codegen/src/printer.rs

```diff
@@ -91,7 +92,11 @@ pub fn print_request_params(
     top_level_statements: &mut TopLevelStatements,
 ) -> String {
     let mut request_parameters = build_request_params(operation);
-    request_parameters.id = query_id;
+    request_parameters.request_format = if let Some(query_id) = &query_id {
+        Some(RequestFormat::ID(query_id.clone()))
+    } else {
+        None
+    };
```

Comment:
Nit, could use `Option::map` here:

```suggestion
    request_parameters.request_format = query_id.map(|id| {
      RequestFormat::ID(id.clone())
    });
```

(untested)